diff --git hello-clean/build-aux/config.sub hello-workdir/build-aux/config.sub
index 90bb8ae..6a932ed 100755
--- hello-clean/build-aux/config.sub
+++ hello-workdir/build-aux/config.sub
@@ -1703,7 +1703,7 @@ case $os in
 	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
-	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
+	     | mirbsd* | netbsd* | nyx* | dicos* | openedition* | ose* \
 	     | bitrig* | openbsd* | solidbsd* | libertybsd* | os108* \
 	     | ekkobsd* | freebsd* | riscix* | lynxos* | os400* \
 	     | bosx* | nextstep* | cxux* | aout* | elf* | oabi* \
